tests/test_list.h

```diff
@@ -45,6 +45,271 @@ static void populate_integers(List<int> &p_list, List<int>::Element *r_elements[
 	}
 }
 
+TEST_CASE("[List] Push/pop back") {
+	List<String> list;
+
+	List<String>::Element *n;
+	n = list.push_back("A");
+	CHECK(n->get() == "A");
+	n = list.push_back("B");
+	CHECK(n->get() == "B");
+	n = list.push_back("C");
+	CHECK(n->get() == "C");
+
+	CHECK(list.size() == 3);
+	CHECK(!list.empty());
+
+	String v;
+	v = list.back()->get();
+	list.pop_back();
+	CHECK(v == "C");
+	v = list.back()->get();
+	list.pop_back();
+	CHECK(v == "B");
+	v = list.back()->get();
+	list.pop_back();
+	CHECK(v == "A");
+
+	CHECK(list.size() == 0);
+	CHECK(list.empty());
+
+	CHECK(list.back() == nullptr);
+	CHECK(list.front() == nullptr);
+}
+
+TEST_CASE("[List] Push/pop front") {
+	List<String> list;
+
+	List<String>::Element *n;
+	n = list.push_front("A");
+	CHECK(n->get() == "A");
+	n = list.push_front("B");
+	CHECK(n->get() == "B");
+	n = list.push_front("C");
+	CHECK(n->get() == "C");
+
+	CHECK(list.size() == 3);
+	CHECK(!list.empty());
+
+	String v;
+	v = list.front()->get();
+	list.pop_front();
+	CHECK(v == "C");
+	v = list.front()->get();
+	list.pop_front();
+	CHECK(v == "B");
+	v = list.front()->get();
+	list.pop_front();
+	CHECK(v == "A");
+
+	CHECK(list.size() == 0);
+	CHECK(list.empty());
+
+	CHECK(list.back() == nullptr);
+	CHECK(list.front() == nullptr);
+}
+
+TEST_CASE("[List] Set and get") {
+	List<String> list;
+	list.push_back("A");
+
+	List<String>::Element *n = list.front();
+	CHECK(n->get() == "A");
+
+	n->set("X");
+	CHECK(n->get() == "X");
+}
+
+TEST_CASE("[List] Insert before") {
+	List<String> list;
+	List<String>::Element *a = list.push_back("A");
+	List<String>::Element *b = list.push_back("B");
+	List<String>::Element *c = list.push_back("C");
+
+	list.insert_before(b, "I");
+
+	CHECK(a->next()->get() == "I");
+	CHECK(c->prev()->prev()->get() == "I");
+	CHECK(list.front()->next()->get() == "I");
+	CHECK(list.back()->prev()->prev()->get() == "I");
+}
+
+TEST_CASE("[List] Insert after") {
+	List<String> list;
+	List<String>::Element *a = list.push_back("A");
+	List<String>::Element *b = list.push_back("B");
+	List<String>::Element *c = list.push_back("C");
+
+	list.insert_after(b, "I");
+
+	CHECK(a->next()->next()->get() == "I");
+	CHECK(c->prev()->get() == "I");
+	CHECK(list.front()->next()->next()->get() == "I");
+	CHECK(list.back()->prev()->get() == "I");
+}
+
+TEST_CASE("[List] Insert before null") {
+	List<String> list;
+	List<String>::Element *a = list.push_back("A");
+	List<String>::Element *b = list.push_back("B");
+	List<String>::Element *c = list.push_back("C");
+
+	list.insert_before(nullptr, "I");
+
+	CHECK(a->next()->get() == "B");
+	CHECK(b->get() == "B");
+	CHECK(c->prev()->prev()->get() == "A");
+	CHECK(list.front()->next()->get() == "B");
+	CHECK(list.back()->prev()->prev()->get() == "B");
+	CHECK(list.back()->get() == "I");
+}
```

Comment:
See also #42116, but this ensures that the current implementation is covered at least.

Reply:
Yeah I think this should likely be fixed.

Reply:
There are two ways to fix this:
1. Modify the behavior so that `insert_before(null)` pushes to the front of the list instead of back.
2. Actually handle this case as an error (for both `insert_before(null)` and `insert_after(null)`), because you can always check whether an element is valid beforehand and decide whether you want to insert a new value elsewhere.